docs/middleware.md

```diff
@@ -377,15 +353,39 @@ class ASGIMiddleware:
         await self.app(scope, receive, send)
 ```
 
-!!! info
```

Comment:
Maybe we can add one of those: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations on the `ASGI3Application` with this mention?

Reply:
I tried that out and it's very subtle to see. I also tried adding a mention such as:

```markdown
* Using [`asgiref`](https://github.com/django/asgiref): for more rigorous type hinting, treating apps as [ASGI v3.0 applications](https://asgi.readthedocs.io/en/latest/specs/main.html#legacy-applications).
```

But in the end I find it more confusing to mention that "hey there's some history behind that `ASGI3*` naming as there used to be an ASGI 2.0 which nobody should be using anymore".

OTOH if the snippet assertively does `import ASGI3Application` and then uses it, then the curious might search for "What is ASGI3", and the others wouldn't be bothered or confused. Thoughts?

Reply:
> I tried that out and it's very subtle to see.

The intention was to be subtle, so only those really curious would click.

In any case, I'm fine with the "look for it on the ASGI documentation if really interested" approach.